azurerm/resource_arm_streamanalytics_job_test.go

```diff
@@ -187,3 +191,181 @@ func TestJSONMarshalling(t *testing.T) {
 	}
 
 }
+
+func TestAccAzureRMStreamAnalyticsJob_outputSqlDb(t *testing.T) {
+
+	ri := acctest.RandInt()
+	config := testAccAzureRMStreamAnalytics_basic(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMStreamAnalyticsJobDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMStreamAnalyticsJobExists("azurerm_streamanalytics_job.test"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMStreamAnalyticsJobDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).streamAnalyticsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_streamanalytics_job" {
+			continue
+		}
+
+		jobName := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(ctx, resourceGroup, jobName, thingsToGet)
+
+		if err != nil {
+			return nil
```

Comment:
Is that what we intended?

Reply:
Yup - I've followed the same pattern established for all other acceptance tests. I too was curious about this and found that the SDK calls have nested calls that propagate up an `err` return value for anything but a 200. 

For this destroy function, this is considered a pass as the resource no longer exists:
Example error (that is effectively swallowed up on the way back):
```
err: streamanalytics.StreamingJobsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'acctestRG-5803807824370763943' could not be found."
```

and the function within the SDK call:
```
func (client StreamingJobsClient) GetResponder(resp *http.Response) (result StreamingJob, err error) {
	err = autorest.Respond(
		resp,
		client.ByInspecting(),
		azure.WithErrorUnlessStatusCode(http.StatusOK),
		autorest.ByUnmarshallingJSON(&result),
		autorest.ByClosing())
	result.Response = autorest.Response{Response: resp}
	return
}
```

Reply:
Cool makes sense